src/terminal/adapter/adaptDispatch.hpp

```diff
@@ -119,6 +124,7 @@ namespace Microsoft::Console::VirtualTerminal
             bool IsOriginModeRelative = false;
             TextAttribute Attributes = {};
             TerminalOutput TermOutput = {};
+            unsigned int CodePage = 0;
```

Comment:
The only part that stuck out to me was ... _Cursor_ state? :smile:

Reply:
Yeah, it's a bit weird, but it's just one of the properties that needs to be saved as part of the _Save Cursor_ operation ([`DECSC`](https://vt100.net/docs/vt510-rm/DECSC.html)). DEC calls this structure the _Cursor Save Buffer_. They don't actually store a code page here, but we need to track the code page in order to save and restore the active GR character set correctly.

Reply:
Totally reasonable.